lib/tasks/errors.rake

```diff
@@ -0,0 +1,41 @@
+namespace :errors do
+  desc <<-END_DESC
+Fetches log for given request id
+
+This task is useful to get log for given request id from logs in log directory. The request_id is required.
+You can also specify log file where request logs has been saved. Default option is 'production.log'.
+
+Available attributes:
+  * request_id = Request id from 500 page
+  * log_file = specify another log file to lookup (default is 'production.log')
+
+Examples:
+  # foreman-rake errors:fetch_log request_id=aabbccss5 [log_file=development.log]
+  END_DESC
+
+  task :fetch_log do
+    request_id = ENV['request_id']
+    logfile = ENV['log_file'] || 'production.log' # default file where we want to find is production.log
+    unless request_id
+      puts "Can't find log without request_id"
+      exit(1)
+    end
+    file_path = File.join(Foreman::Logging.log_directory, logfile)
+    unless File.exist?(file_path)
+      puts "Can't find log file #{file_path}"
+      exit(1)
+    end
+    result = `grep #{request_id} #{file_path}`
+    if result.empty?
+      puts "Can't find log for request_id"
+      exit(1)
+    end
+    puts result
+    puts "\n"
+    puts "Foreman version: #{Foreman::VERSION}"
+    puts "Plugins: "
+    Gem::Specification.all_names.each do |plugin_name|
```

Comment:
This will give us all installed gems, we probably only care about plugins which we can get from `Foreman::Plugin.all` (we would care about `plugin.name` and `plugin.version`)

Reply:
Good point! I didn't know that. It's updated.